documentation/quarkus/guide-exception-handling.asciidoc

```diff
@@ -0,0 +1,65 @@
+:toc:  macro
+toc::[]
+
+= Exception Handling in Quarkus
+
+For link:../guide-exceptions.asciidoc[handling exceptions] within a Spring application, devon4j provides the https://github.com/devonfw/devon4j/tree/master/modules/rest[devon4j-rest module], which provides a https://github.com/devonfw/devon4j/blob/develop/modules/rest/src/main/java/com/devonfw/module/rest/service/impl/RestServiceExceptionFacade.java[`RestServiceExceptionFacade`] to handle all exceptions in a consistent way. Since the module is not suitable for Quarkus, we need to implement this ourselves.
+
+This guide shows how to do that. For an example, see also our https://github.com/devonfw-sample/devon4quarkus-reference[Quarkus reference application].
+
+== Exception mapping
+
+We suggest to implement the exception handling the https://quarkus.io/specs/jaxrs/2.1/index.html#exceptionmapper[JAX-RS way] using `ExceptionMapper<T>`.
+
+RESTEasy provides several https://github.com/quarkusio/quarkus/tree/main/extensions/resteasy-classic/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime[exception mapper] out of the box. For example, RESTEasy's https://github.com/quarkusio/quarkus/blob/main/extensions/resteasy-classic/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NotFoundExceptionMapper.java[`NotFoundExceptionMapper`] provides a web page that shows all available endpoints in dev mode.
+
+However this looks really nice, we want to have consistent exception handling throughout the application.
+We create an abstract class `AbstractExceptionMapper` that acts as base class for all more specific exception mappers, and where the response is created in a consistent manner.
```

Comment:
We create an abstract class `AbstractExceptionMapper` that acts as **a** base class for all **of the** more specific exception mappers and where the response is created in a consistent manner.